File: Data Structures & Algorithms/maximum-product-subarray/submission-1.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int ans = -1e9;
        for(int i=0;i<nums.size();i++){
            int product = 1;
            for(int j=i; j <nums.size();j++){
                product*=nums[j];
                ans = max(ans,product);
                if(product == 0)product = 1;
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-1.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int ans = 0;
        Map<Integer,Integer> freq = new HashMap<>();
        freq.put(0,1);
        int sum = 0;
        for(int i=0;i<nums.length;i++){
            sum += nums[i];
            int diff = sum - k;
            ans += freq.getOrDefault(diff , 0);
            freq.put(sum,freq.getOrDefault(sum,0)+1);
        }
        return ans;
    }
}

